File: user-service/src/main/java/socialapp/userservice/controller/RelationController.java
package socialapp.userservice.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import socialapp.userservice.service.RelationService;

@RestController
@RequestMapping("api/v1/relation")
@RequiredArgsConstructor
public class RelationController {

    private final RelationService relationService;

    @DeleteMapping("/block/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    void deleteBlock(@PathVariable Long id) {
        relationService.deleteBlock(id);
    }

    @DeleteMapping("/subscription/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    void deleteSubscription(@PathVariable Long id) {
        relationService.deleteSubscription(id);
    }

    @DeleteMapping("/friendship/{id}/by/{by}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    void deleteFriendship(@PathVariable Long id, @PathVariable Long by) {
        relationService.deleteFriendship(id, by);
    }

    @PostMapping("/block")
    @ResponseStatus(HttpStatus.CREATED)
    void blockUser(@RequestParam Long sender, @RequestParam Long receiver) {
        relationService.blockUser(sender, receiver);
    }

    @PostMapping("/subscription")
    @ResponseStatus(HttpStatus.CREATED)
    void subscribeUser(@RequestParam Long sender, @RequestParam Long receiver) {
        relationService.subscribeUser(sender, receiver);
    }

    @PostMapping("/friendship")
    @ResponseStatus(HttpStatus.CREATED)
    void addFriendship(@RequestParam Long sender, @RequestParam Long receiver) {
        relationService.addFriendship(sender, receiver);
    }
}

File: user-service/src/main/kotlin/socialapp/ktuserservice/controller/RelationController.kt
package socialapp.ktuserservice.controller

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import socialapp.ktuserservice.model.dto.UserRelationsDto
import socialapp.ktuserservice.service.RelationService


@RestController
@RequestMapping("api/v1/user/relation")
class RelationController(
    private var relationService: RelationService
) {

    @DeleteMapping("/block/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteBlock(@PathVariable id: Long) = relationService.deleteBlock(id)


    @DeleteMapping("/subscription/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteSubscription(@PathVariable id: Long) = relationService.deleteSubscription(id)


    @DeleteMapping("/friendship/{id}/by/{by}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteFriendship(@PathVariable id: Long, @PathVariable by: Long) = relationService.deleteFriendship(id, by)


    @PostMapping("/block")
    @ResponseStatus(HttpStatus.CREATED)
    fun blockUser(@RequestParam sender: Long, @RequestParam receiver: Long) =
        relationService.blockUser(sender, receiver)


    @PostMapping("/subscription")
    @ResponseStatus(HttpStatus.CREATED)
    fun subscribeUser(@RequestParam sender: Long, @RequestParam receiver: Long) =
        relationService.subscribeUser(sender, receiver)


    @PostMapping("/friendship")
    @ResponseStatus(HttpStatus.CREATED)
    fun addFriendship(@RequestParam sender: Long, @RequestParam receiver: Long) =
        relationService.addFriendship(sender, receiver)


    @GetMapping("relations")
    fun userRelations(@RequestParam userId: Long): ResponseEntity<UserRelationsDto> =
        ResponseEntity.ok(relationService.findUserRelations(userId))

}